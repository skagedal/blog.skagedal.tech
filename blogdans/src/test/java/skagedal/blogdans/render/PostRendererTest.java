package skagedal.blogdans.render;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import org.intellij.lang.annotations.Language;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.assertj.core.api.Assertions.*;

@ExtendWith({
    SnapshotExtension.class
})
class PostRendererTest {

    private final PostRenderer postRenderer = new PostRenderer();

    private Expect expect;

    @Test
    void render() {
        final var content = "<p>Hello</p>";
        final var renderedPage = postRenderer.render("First post", content);

        expect.toMatchSnapshot(renderedPage);
    }

    @Disabled
    @Test
    void simple() {
        final var content = "<p>Hello</p>";
        final var renderedPage = postRenderer.render("First post", content);

        final @Language("html") String expectedHtml = """
            <!DOCTYPE html>
            <html>
                 <head>
                      <meta charset="utf-8">
                      <meta http-equiv="X-UA-Compatible" content="IE=edge">
                      <meta name="viewport" content="width=device-width, initial-scale=1">
                      <title>First post</title>
                      <meta name="description" content="Excerpt from First post">
                      <link rel="stylesheet" href="/css/main.css">
                      <link rel="canonical" href="https://blog.skagedal.tech/2025-04-07-first-post/">
                 </head>
                <body>
                    <p>Hello</p>
                </body>
            </html>
            """;
        assertThat(renderedPage)
            .isEqualTo(expectedHtml);
    }
}
