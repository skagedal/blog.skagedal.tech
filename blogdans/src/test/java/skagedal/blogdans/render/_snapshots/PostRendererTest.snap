skagedal.blogdans.render.PostRendererTest.render=[
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <link rel="stylesheet" href="/css/main.css">
        <link rel="canonical" href="https://example.com/2025-04-07-first-post/">
        <title>
            First post
        </title>
    </head>
    <body>
        <header></header>
        <div class="page-content">
            <div class="wrapper">
                <article class="post" itemscope="itemscope" itemtype="http://schema.org/BlogPosting">
                    <header class="post-header">
                        <h1 class="post-title" itemprop="name headline">
                            First post
                        </h1>
                        <p class="post-meta">
                            <time datetime="2025-04-07" itemprop="datePublished">
                                2025-04-07
                            </time>
                        </p>
                    </header>
                    <div class="post-content" itemprop="articleBody">
                        
                    </div>
                </article>
            </div>
        </div>
        <footer></footer>
    </body>
</html>

]


skagedal.blogdans.render.PostRendererTest.withDescription=[
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="description" content="Excerpt from First post">
        <link rel="stylesheet" href="/css/main.css">
        <link rel="canonical" href="https://example.com/2025-04-07-first-post/">
        <title>
            First post
        </title>
    </head>
    <body>
        <header></header>
        <div class="page-content">
            <div class="wrapper">
                <article class="post" itemscope="itemscope" itemtype="http://schema.org/BlogPosting">
                    <header class="post-header">
                        <h1 class="post-title" itemprop="name headline">
                            First post
                        </h1>
                        <p class="post-meta">
                            <time datetime="2025-04-07" itemprop="datePublished">
                                2025-04-07
                            </time>
                        </p>
                    </header>
                    <div class="post-content" itemprop="articleBody">
                        
                    </div>
                </article>
            </div>
        </div>
        <footer></footer>
    </body>
</html>

]